include/dataformats/pacman/PACMANFrame.hpp

```diff
@@ -57,163 +49,70 @@ class PACMANFrame
-  }
-
-  static constexpr int PACKET_CHANNELID_OFFSET = 10;                // bits
-  static constexpr std::uint_fast8_t PACKET_CHANNELID_MASK{ 0x3F }; // bitmask
-  uint64_t get_packet_channelid(uint64_t* packet)
-  {
-    // bits [10:15], only valid for data packets
-    return get_packet_data(packet, PACKET_CHANNELID_OFFSET, PACKET_CHANNELID_MASK);
-  }
-
-  static constexpr int PACKET_TIMESTAMP_OFFSET = 16;                       // bits
-  static constexpr std::uint_fast32_t PACKET_TIMESTAMP_MASK{ 0x7FFFFFFF }; // bitmask
-  uint64_t get_packet_timestamp(uint64_t* packet)
-  {
-    // bits [46:16], only valid for data packets
-    return get_packet_data(packet, PACKET_TIMESTAMP_OFFSET, PACKET_TIMESTAMP_MASK);
-  }
-
-  static constexpr int PACKET_FIRST_PACKET_OFFSET = 47;               // bits
-  static constexpr std::uint_fast8_t PACKET_FIRST_PACKET_MASK{ 0x1 }; // bitmask
-  uint64_t get_packet_first_packet(uint64_t* packet)
-  {
-    // bits [47], only valid for data packets
-    return get_packet_data(packet, PACKET_FIRST_PACKET_OFFSET, PACKET_FIRST_PACKET_MASK);
-  }
-
-  static constexpr int PACKET_DATAWORD_OFFSET = 48;                // bits
-  static constexpr std::uint_fast8_t PACKET_DATAWORD_MASK{ 0xFF }; // bitmask
-  uint64_t get_packet_dataword(uint64_t* packet)
-  {
-    // bits [48:55], only valid for data packets
-    return get_packet_data(packet, PACKET_DATAWORD_OFFSET, PACKET_DATAWORD_MASK);
-  }
-
-  static constexpr int PACKET_TRIGGER_TYPE_OFFSET = 56;               // bits
-  static constexpr std::uint_fast8_t PACKET_TRIGGER_TYPE_MASK{ 0x3 }; // bitmask
-  uint64_t get_packet_trigger_type(uint64_t* packet)
-  {
-    // bits [56:57], only valid for data packets
-    return get_packet_data(packet, PACKET_TRIGGER_TYPE_OFFSET, PACKET_TRIGGER_TYPE_MASK);
-  }
+    msg_type type:8; // message type indicator
+    uint32_t unix_ts; // unix timestamp from PACMAN dataserver
+    uint8_t _null; // unused
+    uint16_t words; // number of words in message
+  };
 
-  static constexpr int PACKET_LOCAL_FIFO_STATUS_OFFSET = 58;               // bits
-  static constexpr std::uint_fast8_t PACKET_LOCAL_FIFO_STATUS_MASK{ 0x3 }; // bitmask
-  uint64_t get_packet_local_fifo_status(uint64_t* packet)
-  {
-    // bits [58:59], only valid for data packets
-    return get_packet_data(packet, PACKET_LOCAL_FIFO_STATUS_OFFSET, PACKET_LOCAL_FIFO_STATUS_MASK);
-  }
+  union LArPixPacket
+  {
+    struct __attribute__((__packed__)) LArPixGenericPacket
+    { // A generic type applicable to all LArPix packets
+      packet_type type:2; // packet type indicator
+      uint64_t _null:62; // packet data (specific for each packet type)
+    } packet;
+
+    struct __attribute__((__packed__)) LArPixDataPacket
+    { // Specific for LArPix data packets
+      packet_type type:2; // packet type indicator
+      uint8_t chipid:8; // asic chip id
+      uint8_t channelid:6; // asic channel id
```

Comment:
If you want to get rid of the note from GCC, you can replace the types of these three fields with `uint16_t`, and it should have the same arrangement in memory.